src/components/ColorPalette/ColorPalette.js

```diff
@@ -1,92 +1,164 @@
 import React from 'react';
+import { connect } from 'react-redux';
 import PropTypes from 'prop-types';
 
 import Icon from 'components/Icon';
 
 import getBrightness from 'helpers/getBrightness';
+import selectors from 'selectors';
 
 import './ColorPalette.scss';
 
+const dataElement = 'colorPalette';
+
 class ColorPalette extends React.PureComponent {
   static propTypes = {
     property: PropTypes.string.isRequired,
     color: PropTypes.object.isRequired,
     onStyleChange: PropTypes.func.isRequired,
-  }
-
-  constructor(props) {
-    super(props);
-    this.palette = [
-      ['#F1A099', '#FFC67B', '#FFE6A2', '#80E5B1', '#92E8E8', '#A6A1E6', '#E2A1E6'],
-      ['#E44234', '#FF8D00', '#FFCD45', '#00CC63', '#25D2D1', '#4E7DE9', '#C544CE'],
-      ['#88271F', '#B54800', '#F69A00', '#007A3B', '#167E7D', '#2E4B8B', '#76287B'],
-      [null, '#FFFFFF', '#CDCDCD', '#9C9C9C', '#696969', '#373737', '#000000'],
-    ];
-  }
+    overridePalette: PropTypes.array,
+  };
+
+  defaultPalette = [
+    '#F1A099',
+    '#FFC67B',
+    '#FFE6A2',
+    '#80E5B1',
+    '#92E8E8',
+    '#A6A1E6',
+    '#E2A1E6',
+    '#E44234',
+    '#FF8D00',
+    '#FFCD45',
+    '#00CC63',
+    '#25D2D1',
+    '#4E7DE9',
+    '#C544CE',
+    '#88271F',
+    '#B54800',
+    '#F69A00',
+    '#007A3B',
+    '#167E7D',
+    '#2E4B8B',
+    '#76287B',
+    'transparency',
+    '#FFFFFF',
+    '#CDCDCD',
+    '#9C9C9C',
+    '#696969',
+    '#373737',
+    '#000000',
+  ];
 
   setColor = e => {
     const { property, onStyleChange } = this.props;
     const bg = e.target.style.backgroundColor; // rgb(r, g, b);
-    const rgba = bg ? bg.slice(bg.indexOf('(') + 1, -1).split(',') : [0, 0, 0, 0];
-    const color = new window.Annotations.Color(rgba[0], rgba[1], rgba[2], rgba[3]);
+    const rgba = bg
+      ? bg.slice(bg.indexOf('(') + 1, -1).split(',')
+      : [0, 0, 0, 0];
+    const color = new window.Annotations.Color(
+      rgba[0],
+      rgba[1],
+      rgba[2],
+      rgba[3],
+    );
     onStyleChange(property, color);
-  }
+  };
 
-  renderTransparencyCell = (bg, key) => {
+  renderTransparencyCell = bg => {
     const { property } = this.props;
-    const shouldRenderDummyCell = property === 'TextColor' || property === 'StrokeColor';
+    const shouldRenderDummyCell =
+      property === 'TextColor' || property === 'StrokeColor';
 
     if (shouldRenderDummyCell) {
-      return <div className="dummy-cell" key={key}></div>;
+      return <div className="dummy-cell" />;
```

Comment:
i'm still getting the hang of react.
what is the difference between the two? I'm guessing one just doesn't need the key? but both renders it as an HTML element?

Reply:
These two doesn't have any difference. The new one is a shortcut. We can use the shortcut if the element we are rendering doesn't have a child.

I didn't use the `key` for this `div` because I refactored the code to use key here https://github.com/PDFTron/webviewer-ui/blob/master/src/components/Input/Input.js#L17.